src/main/java/com/training/springpractice/controllers/CatalogController.java

```diff
@@ -0,0 +1,64 @@
+package com.training.springpractice.controllers;
+
+import com.training.springpractice.models.Catalog;
+import com.training.springpractice.models.CatalogBody;
+import com.training.springpractice.services.CatalogService;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/v1/catalog-entry")
+public class CatalogController {
+
+    @Autowired
+    private CatalogService service;
+
+    private ModelMapper mapper;
+
+    @PostMapping
+    public ResponseEntity create(@Valid @RequestBody CatalogBody body) {
+        try {
+            Catalog entity = convertToEntity(body);
+            Catalog createdCatalog = service.create(entity);
+            CatalogBody response = convertToDto(createdCatalog);
+            return ResponseEntity.ok(response);
+        } catch (Exception e) {
+            RuntimeException apiError = new RuntimeException(e.getMessage());
+            return new ResponseEntity(apiError, new HttpHeaders(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @Autowired
```

Comment:
Why you are using different types of injection in the same file? It is a good practice to mantain/use the same pattern in order your code is understandable, if you have to change this type of injection to another one, this will be still working fine?